test/unit/range/view/view_concept_check.hpp

```diff
@@ -0,0 +1,295 @@
+        {
+            debug_stream << "Weak lost check of concept '" << to_string(*it) << "' failed:\n" 
+                         << "OUT_TYPE: " << fulfilled<OUT_TYPE>(*it) << " (expected false)" <<'\n';
+            success = false;
+        }
+    }
+    return success;
+}
+
+/*!\brief Checks if the fulfillment of a concept is guaranteed from IN_TYPE to OUT_TYPE.
+ * \tparam concepts    A list of range concepts to check as seqan3::test::ConceptType object.
+ * \return True if fulfillment of conc is guaranteed in OUT_TYPE. False if not.
+ *
+ * ### Example
+ * using namespace seqan3::test;
+ * std::string vec{"HelloWorld"};
+ * auto view = vec | std::view::reverse;
+ * // This EXPECT_TRUE will fail, as InputRange concept is fulfilled by IN_TYPE.
+ * // For this setup, you would have to use weak_guaranteed(...) function instead.
+ * EXPECT_TRUE((guaranteed<decltype(vec),decltype(view)>({Input}))); // double brackets required
+ *
+ * \details
+ *
+ * The lost function checks a real guarantee, i.e. false for IN_TYPE and true for OUT_TYPE. 
+ * You can use weak_guarantee(...) function to only check true for OUT_TYPE.
+ * This function is designed for use inside EXPECT_TRUE(...). Using other test scenarios (e.g., EXPECT_FALSE)
+ * can lead to unintended command line output via debug_stream.
+ */
+template<typename IN_TYPE, typename OUT_TYPE>
+constexpr bool guaranteed(std::initializer_list<ConceptType> const & concepts)
+{
+    bool success = true;
+    auto it = concepts.begin();
+    for (; it != concepts.end(); ++it)
+    {
+        if(fulfilled<IN_TYPE>(*it) || !fulfilled<OUT_TYPE>(*it))
+        {
+            debug_stream << "Guaranteed check of concept '" << to_string(*it) << "' failed:\n" 
+                         << " IN_TYPE: " << fulfilled<IN_TYPE>(*it) << " (expected false)" << '\n'
+                         << "OUT_TYPE: " << fulfilled<OUT_TYPE>(*it) << " (expected true)" <<'\n';
+            success = false;
+        }
+    }
+    return success;
+}
+
+/*!\brief Checks weakly if the fulfillment of a concept is lost from IN_TYPE to OUT_TYPE.
+ * \tparam concepts    A list of range concepts to check as seqan3::test::ConceptType object.
+ * \return True if fulfillment of conc is (weakly) guaranteed in OUT_TYPE. False if not.
+ *
+ * ### Example
+ * using namespace seqan3::test;
+ * std::string vec{"HelloWorld"};
+ * auto view = vec | std::view::reverse;
+ * EXPECT_TRUE(weak_guaranteed<decltype(view)>({Input}));
+ *
+ * \details
+ *
+ * The weak_guaranteed(...) function weakly checks a guarantee, i.e. only true for OUT_TYPE. 
+ * You can use guaranteed(...) function to check true for IN_TYPE and true for OUT_TYPE.
+ * This function is designed for use inside EXPECT_TRUE(...). Using other test scenarios (e.g., EXPECT_FALSE)
+ * can lead to unintended command line output via debug_stream.
+ */
+template<typename OUT_TYPE>
+constexpr bool weak_guaranteed(std::initializer_list<ConceptType> const & concepts)
```

Comment:
```suggestion
constexpr bool weakly_guaranteed(std::initializer_list<ConceptType> const & concepts)
```
I'd say it's an adverb in this case.
(How is it guaranteed? Weakly.)

Reply:
Nope, it's a weak test of the guarantee ;)
I don't think this should be an adverb, there is nothing like a weak guarantee so this would be confusing.